Entrypoints don't source /var/lib/generacy-app-config/env; worker doesn't mount the app-config volume — user-configured env vars never reach orchestrator or workflow processes

## Summary

App-config env vars set via the bootstrap wizard / Settings panel are correctly persisted by the control-plane daemon to `/var/lib/generacy-app-config/env` (non-secret) and the encrypted `ClusterLocalBackend` (secret, will be rendered to `/run/generacy-app-config/secrets.env` by generacy-ai/generacy#632). But the cluster's entrypoint scripts never source either file, so:

- Orchestrator and its child processes (the control-plane daemon, generacy CLI, any spawned workflow runner) don't see the env vars.
- Workers don't see them either, and the worker service in compose doesn't even mount the `generacy-app-config-data` volume — workers cannot read the env file at all.
- Agent workflows spawned in workers (Claude Code, MCP servers, etc.) inherit only the entrypoint-set env, so they can't read user-configured env vars like `LIVEKIT_URL`, `SERVICE_ANTHROPIC_API_KEY`.

The motivating use case (a LiveKit voice agent project) needs `LIVEKIT_URL`, `LIVEKIT_API_KEY`, `SERVICE_ANTHROPIC_API_KEY` etc. available to agent workflows and to any user services compose'd into the cluster.

Verified on staging on 2026-05-15: values are in `/var/lib/generacy-app-config/env` and `values.yaml`, but `cat /proc/1/environ` on both orchestrator and worker shows none of them.

## Scope

### 1. Mount the app-config volume on the worker service (read-only)

`docker-compose.yml` worker service is missing the volume mount. Add:

```yaml
worker:
  volumes:
    # ... existing mounts ...
    - generacy-app-config-data:/var/lib/generacy-app-config:ro
```

Read-only because only the orchestrator's control-plane daemon writes; workers consume.

### 2. Add tmpfs for the secrets file on both services

Once generacy-ai/generacy#632 lands, the control-plane daemon renders secrets to `/run/generacy-app-config/secrets.env`. Both orchestrator (which writes it) and worker (which reads it) need the tmpfs mount:

```yaml
orchestrator:
  tmpfs:
    - /run/generacy-control-plane:mode=1750,uid=1000,gid=1000
    - /run/generacy-app-config:mode=1750,uid=1000,gid=1000   # NEW

worker:
  tmpfs:
    - /run/generacy-credhelper:mode=1750,uid=1002,gid=1000
    - /run/generacy-control-plane:mode=1750,uid=1000,gid=1000
    - /run/generacy-app-config:mode=1750,uid=1000,gid=1000   # NEW
```

(Workers don't write to secrets.env directly — they consume what the orchestrator's control-plane wrote. But each worker still needs its own tmpfs because secret content shouldn't traverse a shared host path; the daemon should push secret renders to the workers on update. For v1, simpler: re-render on PUT only writes to the orchestrator's tmpfs; workers source it through a mechanism TBD — see Open question below.)

### 3. Source both env files in the orchestrator entrypoint

In `.devcontainer/generacy/scripts/entrypoint-orchestrator.sh`, after the existing wizard-credentials sourcing block (which handles `GH_TOKEN` etc.), add:

```bash
# Source app-config env vars set via the bootstrap wizard / Settings panel.
# Non-secret: /var/lib/generacy-app-config/env (persistent volume)
# Secret:     /run/generacy-app-config/secrets.env (tmpfs, re-rendered at boot from
#             encrypted backend by the control-plane daemon — see generacy#632)
for app_env in /var/lib/generacy-app-config/env /run/generacy-app-config/secrets.env; do
  if [ -f "$app_env" ]; then
    log "Sourcing app-config env from $app_env"
    set -a
    # shellcheck disable=SC1090
    source "$app_env"
    set +a
  fi
done
```

Mirror the same block in `entrypoint-worker.sh` (and `entrypoint-post-activation.sh` if that's a separate process surface that should see the values too).

## Open question — workers and secrets

Approach A (current draft): worker tmpfs is independent of orchestrator tmpfs. Some mechanism propagates the rendered secrets.env from orchestrator to workers (control-plane could write via a per-worker socket, or workers could pull on demand). Adds complexity.

Approach B (simpler): drop secret env-var support for workers entirely — secrets reach workflows via the credhelper's per-session render path, which already works for workflows. Non-secret env vars still propagate via the read-only mount. Workers that need long-lived secrets are an edge case in v1.

Recommend B for v1, document A as future work if workloads emerge that need it. This issue should ship the non-secret path immediately and stage the secret path behind generacy-ai/generacy#632.

## Test plan
- [ ] Build image with the changes; `docker exec <worker> ls /var/lib/generacy-app-config/env` returns the file content.
- [ ] After setting `LIVEKIT_URL` via the UI, restart the orchestrator and `docker exec <orchestrator> sh -c 'cat /proc/1/environ | tr "\0" "\n" | grep LIVEKIT'` shows the value.
- [ ] Same check on a worker container after restart.
- [ ] Agent workflows spawned from the worker have the env var in their process environment.
- [ ] Once generacy#632 lands and the tmpfs mount is added, secret env vars are similarly visible on the orchestrator after restart.

## Related
- generacy-ai/generacy#632 — renders secrets to `/run/generacy-app-config/secrets.env`. Required for the secret-env path to work.
- generacy-ai/generacy#622 — original feature; documented "entrypoint sources the materialized env file" as intent but the implementation didn't land in the cluster image.
- generacy-ai/generacy-cloud#583 — UI side.
- cluster-microservices will pick this up via the usual cluster-base sync flow once merged.
- Reported during app-config testing on 2026-05-15.

--- a/.devcontainer/generacy/docker-compose.yml
+++ b/.devcontainer/generacy/docker-compose.yml
@@ -28,6 +28,11 @@ services:
     tmpfs:
       # in-cluster control-plane HTTP service socket, owned by the node uid
       - /run/generacy-control-plane:mode=1750,uid=1000,gid=1000
+      # Rendered secret app-config env vars (`secrets.env`). Written by the
+      # control-plane daemon on PUT and re-rendered at boot from the encrypted
+      # ClusterLocalBackend — see generacy-ai/generacy#632. Tmpfs because the
+      # plaintext rendering must never hit disk.
+      - /run/generacy-app-config:mode=1750,uid=1000,gid=1000
     stop_grace_period: 30s
     volumes:
       # Clone mode: persistent volume avoids host mount issues (especially on Windows)
@@ -103,6 +108,14 @@ services:
       - /run/generacy-credhelper:mode=1750,uid=1002,gid=1000
       # in-cluster control-plane HTTP service socket, owned by the node uid
       - /run/generacy-control-plane:mode=1750,uid=1000,gid=1000
+      # Rendered secret app-config env vars (`secrets.env`). Mirrors the
+      # orchestrator tmpfs so worker-side processes have the same surface for
+      # sourcing app-config secrets — keeps dev/testing parity between
+      # orchestrator and worker. Propagation of secret content from the
+      # orchestrator's control-plane to per-worker tmpfs is TBD (see issue #38
+      # Open question / generacy-ai/generacy#632); the tmpfs is in place ahead
+      # of that wiring.
+      - /run/generacy-app-config:mode=1750,uid=1000,gid=1000
     stop_grace_period: 30s
     deploy:
       replicas: ${WORKER_COUNT:-3}
@@ -117,6 +130,11 @@ services:
       # the named volume on first-init. The credhelper uid (1002) writes
       # master.key inside as 0600 owned by 1002.
       - generacy-data:/var/lib/generacy
+      # App-config state dir (read-only on workers): orchestrator's control-plane
+      # daemon writes the materialized non-secret env file
+      # (/var/lib/generacy-app-config/env) and metadata YAML; workers consume.
+      # Read-only because only the orchestrator writes.
+      - generacy-app-config-data:/var/lib/generacy-app-config:ro
     env_file:
       - path: .env
       - path: .env.local

--- a/.devcontainer/generacy/scripts/entrypoint-orchestrator.sh
+++ b/.devcontainer/generacy/scripts/entrypoint-orchestrator.sh
@@ -23,6 +23,23 @@ if [ -f "$WIZARD_CREDS" ]; then
     set +a
 fi
 
+# Source app-config env vars set via the bootstrap wizard / Settings panel so
+# orchestrator + child processes (control-plane daemon, generacy CLI, spawned
+# workflow runners) inherit user-configured values like LIVEKIT_URL.
+#   - /var/lib/generacy-app-config/env       non-secret, persistent volume
+#   - /run/generacy-app-config/secrets.env   secret, tmpfs, re-rendered at boot
+#     from the encrypted ClusterLocalBackend by the control-plane daemon
+#     (see generacy-ai/generacy#632)
+for app_env in /var/lib/generacy-app-config/env /run/generacy-app-config/secrets.env; do
+    if [ -f "$app_env" ]; then
+        log "Sourcing app-config env from $app_env"
+        set -a
+        # shellcheck disable=SC1090
+        source "$app_env"
+        set +a
+    fi
+done
+
 # Configure git credentials
 bash /usr/local/bin/setup-credentials.sh
 

--- a/.devcontainer/generacy/scripts/entrypoint-post-activation.sh
+++ b/.devcontainer/generacy/scripts/entrypoint-post-activation.sh
@@ -43,6 +43,18 @@ if [ -f "$WIZARD_CREDS" ]; then
     set +a
 fi
 
+# Source app-config env vars so the post-activation generacy setup sees
+# user-configured values. Mirrors the same block in entrypoint-orchestrator.sh.
+for app_env in /var/lib/generacy-app-config/env /run/generacy-app-config/secrets.env; do
+    if [ -f "$app_env" ]; then
+        log "Sourcing app-config env from $app_env"
+        set -a
+        # shellcheck disable=SC1090
+        source "$app_env"
+        set +a
+    fi
+done
+
 # Step 1: configure git/gh credentials from the env vars the wizard delivered
 bash /usr/local/bin/setup-credentials.sh
 

--- a/.devcontainer/generacy/scripts/entrypoint-worker.sh
+++ b/.devcontainer/generacy/scripts/entrypoint-worker.sh
@@ -25,6 +25,25 @@ fi
 
 log "Starting worker setup..."
 
+# Source app-config env vars set via the bootstrap wizard / Settings panel so
+# worker + child processes (agent workflows, MCP servers, user services)
+# inherit user-configured values like LIVEKIT_URL, SERVICE_ANTHROPIC_API_KEY.
+#   - /var/lib/generacy-app-config/env       non-secret, RO mount from named
+#     volume the orchestrator's control-plane writes
+#   - /run/generacy-app-config/secrets.env   secret, per-worker tmpfs. Empty
+#     in v1 — propagation from the orchestrator is TBD (see issue #38 Open
+#     question / generacy-ai/generacy#632); guarded by -f so the no-file case
+#     is a silent no-op.
+for app_env in /var/lib/generacy-app-config/env /run/generacy-app-config/secrets.env; do
+    if [ -f "$app_env" ]; then
+        log "Sourcing app-config env from $app_env"
+        set -a
+        # shellcheck disable=SC1090
+        source "$app_env"
+        set +a
+    fi
+done
+
 # Configure git credentials
 bash /usr/local/bin/setup-credentials.sh
 

